lego/pytest_lego/plugin.py

```diff
@@ -3,45 +3,122 @@
-    for our test wrapper to create components.
-    Using the lego manager API the test wrapper acquires the components and
-    provide them to the original test function.
-
     Args:
-        request: A PyTest fixture helper.
+        request: A PyTest fixture helper, with information on the requesting test function.
 
     Returns:
         RPyC connection to LegoManager service.
     """
 
-    assert MARK in request.config.inicfg.config.sections, f'Missing {MARK} section in inifile'
-    assert 'lego_manager' in request.config.inicfg.config.sections[MARK], (
-        'Missing lego_manager hostname in inifile')
+    assert LEGO_MARK in request.config.inicfg.config.sections, f'Missing {LEGO_MARK} section in inifile'
+
+    try:
+        manager_hostname = request.config.inicfg.config.sections[LEGO_MARK]['lego_manager_hostname']
+        manager_port = request.config.inicfg.config.sections[LEGO_MARK]['lego_manager_port']
+    except KeyError as e:
+        missing_key = e.args[0]
+        raise KeyError(f'Missing {missing_key} under {LEGO_MARK} section in inifile')
 
-    manager = request.config.inicfg.config.sections[MARK]['lego_manager']
-    lego_manager = rpyc.connect(host=manager, port=18861)
+    lego_manager = rpyc.connect(manager_hostname, manager_port)
     request.addfinalizer(lego_manager.close)
 
     return lego_manager
 
 
+@contextlib.contextmanager
+def get_components(request, lego_manager) -> List[BaseComponent]:
+    """Provides the components requested in corresponding lego mark for the test.
+
+    This fixture provides the components requested by the test function.
+    To provide arguments to the components, one should define [component] section in pytest config file.
+
+    Typical usage example:
+
+    @pytest.mark.lego('zebra.alice and elephant.bob')
+    def test_default_sizes(self, components):
+        zebra, elephant, *_ = components
+        assert zebra.size == elephant.size
+
+    Example for configuration in pytest.ini:
+    [zebra.alice]
+    hostname = zebra
+    username = admin
+    password = password
+    [elephant.bob]
+    hostname=192.168.100.11
+    name = Bob
+    size = 120
+
+    Args:
+        request: A PyTest fixture helper, with information on the requesting test function.
+        lego_manager: An RPyC connection to LegoManager service.
+
+    Returns:
+        List of components requested in lego.mark.
+    """
+    lego_mark = request.node.get_closest_marker(LEGO_MARK)
+    if lego_mark is None:
+        # The test doesn't have the 'lego' mark.
+        yield None
+        # There is no resources to free.
+        return
+
+    with component_factory.acquire_components(
+            lego_manager,
+            request.config,
+            *lego_mark.args,
+            **lego_mark.kwargs
+    ) as components:
+        yield components
+
+
+@pytest.fixture(scope='function')
+def components_func(request, lego_manager) -> List[BaseComponent]:
```

Comment:
I don't really like the naming of components_func nad components_cls.
Maybe you can use the class scope in both cases?
If not rename `components_func` to `components` and keep `components_cls` as is.